client/tx/factory_test.go

```diff
@@ -1,35 +1,117 @@
-package tx_test
+package tx
 
 import (
 	"testing"
 
 	"github.com/stretchr/testify/require"
 
 	"github.com/cosmos/cosmos-sdk/client"
-	"github.com/cosmos/cosmos-sdk/client/tx"
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
+	"github.com/cosmos/cosmos-sdk/crypto/hd"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/multisig"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	"github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/cosmos/cosmos-sdk/testutil/testdata"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
 )
 
-func TestFactoryPrepate(t *testing.T) {
+func TestFactoryPrepare(t *testing.T) {
 	t.Parallel()
 
-	factory := tx.Factory{}
+	factory := Factory{}
 	clientCtx := client.Context{}
 
 	output, err := factory.Prepare(clientCtx.WithOffline(true))
 	require.NoError(t, err)
 	require.Equal(t, output, factory)
 
-	factory = tx.Factory{}.WithAccountRetriever(client.MockAccountRetriever{ReturnAccNum: 10, ReturnAccSeq: 1}).WithAccountNumber(5)
+	factory = Factory{}.WithAccountRetriever(client.MockAccountRetriever{ReturnAccNum: 10, ReturnAccSeq: 1}).WithAccountNumber(5)
 	output, err = factory.Prepare(clientCtx.WithFrom("foo"))
 	require.NoError(t, err)
 	require.NotEqual(t, output, factory)
 	require.Equal(t, output.AccountNumber(), uint64(5))
 	require.Equal(t, output.Sequence(), uint64(1))
 
-	factory = tx.Factory{}.WithAccountRetriever(client.MockAccountRetriever{ReturnAccNum: 10, ReturnAccSeq: 1})
+	factory = Factory{}.WithAccountRetriever(client.MockAccountRetriever{ReturnAccNum: 10, ReturnAccSeq: 1})
 	output, err = factory.Prepare(clientCtx.WithFrom("foo"))
 	require.NoError(t, err)
 	require.NotEqual(t, output, factory)
 	require.Equal(t, output.AccountNumber(), uint64(10))
 	require.Equal(t, output.Sequence(), uint64(1))
 }
+
+func TestFactory_getSimPK(t *testing.T) {
+	// setup keyring
+	registry := codectypes.NewInterfaceRegistry()
+	cryptocodec.RegisterInterfaces(registry)
+	k := keyring.NewInMemory(codec.NewProtoCodec(registry))
+
+	tests := []struct {
+		name     string
+		fromName string
+		genKey   func(fromName string, k keyring.Keyring) error
+		want     types.PubKey
+	}{
+		{
+			name:     "simple key",
+			fromName: "testKey",
+			genKey: func(fromName string, k keyring.Keyring) error {
+				_, err := k.NewAccount(fromName, testdata.TestMnemonic, "", "", hd.Secp256k1)
+				return err
+			},
+			want: &secp256k1.PubKey{},
+		},
+		{
+			name:     "multisig key",
+			fromName: "multiKey",
+			genKey: func(fromName string, k keyring.Keyring) error {
+				pk := multisig.NewLegacyAminoPubKey(1, []types.PubKey{&multisig.LegacyAminoPubKey{}})
+				_, err := k.SaveMultisig(fromName, pk)
+				return err
+			},
+			want: &multisig.LegacyAminoPubKey{},
```

Comment:
You can use `wantType: (*multisig.LegacyAminoPubKey)(nil)` to make sure that future maintainers don't get caught up in trying to see if they need a populated value, meanwhile we are just comparing types.